mtags/src/main/scala-3/scala/meta/internal/pc/PcDefinitionProvider.scala

```diff
@@ -0,0 +1,197 @@
+import dotty.tools.dotc.core.StdNames
+import dotty.tools.dotc.core.Symbols._
+import dotty.tools.dotc.core.Types.NamedType
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.interactive.SourceTree
+import dotty.tools.dotc.transform.SymUtils._
+import dotty.tools.dotc.util.NoSourcePosition
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.SourcePosition
+import org.eclipse.lsp4j.Location
+
+class PcDefinitionProvider(
+    driver: InteractiveDriver,
+    params: OffsetParams,
+    search: SymbolSearch
+) {
+
+  def definitions(): DefinitionResult = {
+    val uri = params.uri
+    val filePath = Paths.get(uri)
+    val diagnostics = driver.run(
+      uri,
+      SourceFile.virtual(filePath.toString, params.text)
+    )
+    val unit = driver.currentCtx.run.units.head
+    val tree = unit.tpdTree
+
+    val pos = driver.sourcePosition(params)
+    val path =
+      Interactive.pathTo(driver.openedTrees(uri), pos)(using driver.currentCtx)
+    given ctx: Context = driver.localContext(params)
+    val indexedContext = IndexedContext(ctx)
+    findDefinitions(tree, path, pos, driver, indexedContext)
+  }
+
+  private def findDefinitions(
+      tree: Tree,
+      path: List[Tree],
+      pos: SourcePosition,
+      driver: InteractiveDriver,
+      indexed: IndexedContext
+  ): DefinitionResult = {
+    import indexed.ctx
+    enclosingSymbols(path, pos, indexed) match {
+      case symbols @ (sym :: other) =>
+        val isLocal = sym.source == pos.source
+        if (isLocal)
+          val defs =
+            Interactive.findDefinitions(List(sym), driver, false, false)
+          defs.headOption match {
+            case Some(srcTree) =>
+              val pos = srcTree.namePos
+              pos.toLocation match {
+                case None => DefinitionResultImpl.empty
+                case Some(loc) =>
+                  DefinitionResultImpl(
+                    SemanticdbSymbols.symbolName(sym),
+                    List(loc).asJava
+                  )
+              }
+            case None =>
+              DefinitionResultImpl.empty
+          }
+        else {
+          val res = new ArrayList[Location]()
+          semanticSymbolsSorted(symbols)
+            .foreach { sym =>
+              res.addAll(search.definition(sym, params.uri()))
+            }
+          DefinitionResultImpl(
+            SemanticdbSymbols.symbolName(sym),
+            res
+          )
+        }
+      case Nil => DefinitionResultImpl.empty
+    }
+  }
+
+  @tailrec
+  private def enclosingSymbols(
+      path: List[Tree],
+      pos: SourcePosition,
+      indexed: IndexedContext
+  ): List[Symbol] = {
+    import indexed.ctx
+    path match {
+      // For a named arg, find the target `DefDef` and jump to the param
+      case NamedArg(name, _) :: Apply(fn, _) :: _ =>
+        val funSym = fn.symbol
+        if (
+          funSym.name == StdNames.nme.copy
```

Comment:
```suggestion
          funSym.name == StdNames.nme.copy
```
what about `apply` ? Wouldn't the two next conditions be enough?

Reply:
Good catch! Updated and added tests for that.
Honestly, as it's a it was initially copypasted from `Interactive` I just ignored these lines :smile: